app/queries/create_vimeo_video_mutator.rb

```diff
@@ -0,0 +1,34 @@
+class CreateVimeoVideoMutator < ApplicationQuery
+  include AuthorizeAuthor
+  
+  property :target_id, validates: { presence: true }
+  property :size, validates: { presence: true }
+  
+  def create_vimeo_video
+    path = '/me/videos'
+    data = {
+      upload: {
+        approach: 'tus',
+        size: size
+      },
+      privacy: {
+        embed: 'whitelist'
+      }
+    }
+
+    api_service = Vimeo::ApiService.new(path, current_school)
+    response = api_service.post(data)
+
+    raise StandardError if response[:error_code]
```

Comment:
Let's preserve info about the error in the raised exception:

```suggestion
    raise "Encountered error with code #{response[:error_code]} when trying to create a Vimeo video." if response[:error_code].present?
```